app/lib/account/agent.dart

```diff
@@ -12,41 +12,74 @@ import 'models.dart';
 final _uuidRegExp =
     RegExp(r'^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$');
 
+/// Agents identify authenticated users or automated system accounts.
+///
+/// The format and/or the content prefix identifies the type of the agent.
+/// Newer agent identifiers must contain a unique (but maybe opaque)
+/// part that corresponds with an internal or external authentication
+/// identifier (like email address or repository URL).
+///
+/// Older identifiers may only describe the type of the agent.
 abstract class KnownAgents {
-  static const githubActions = 'service:github-actions';
-  static const gcpServiceAccount = 'service:gcp-service-account';
+  /// Non-specific agent - only specifies it is from GitHub Actions.
+  static const genericGithubActions = 'service:github-actions';
+
+  /// Non-specific agent - only specifies it is from GCP Service Account.
+  static const genericGcpServiceAccount = 'service:gcp-service-account';
+
   // Agent for pub admin actions.
   static const pubSupport = 'support@pub.dev';
 
-  static const _values = <String>{
-    githubActions,
-    gcpServiceAccount,
+  /// Returns an agentId in the format of `service:github-actions:<repositoryOwnerId>/<repositoryId>`
+  static String githubActionsAgentId({
```

Comment:
```suggestion
  /// Returns an agentId in the format of `service:github-actions:<repositoryOwnerId>/<repositoryId>`
  ///
  /// Note: [repositoryOwnerId] and [repositoryId] are opaque identifiers.
  /// These are **NOT** the same as the repository name, these identifiers
  /// remain stable when organizations and repositories are renamed!
  static String githubActionsAgentId({
```

Just to reduce risk that someone plugs in the wrong value.